libsel4debug/src/bootinfo.c

```diff
@@ -8,41 +8,112 @@
 #include <sel4debug/gen_config.h>
 
 #include <stdio.h>
+#include <assert.h>
 
 #include <sel4/sel4.h>
 #include <utils/util.h>
 
+static void print_slot_reg_info(seL4_SlotRegion *reg)
+{
+    if (reg->end == reg->start) {
+        printf("none ");
+    }
+    printf("[%"SEL4_PRIu_word" --> %"SEL4_PRIu_word"]",
+           reg->start, reg->end);
+}
+
 void debug_print_bootinfo(seL4_BootInfo *info)
 {
+    printf("Node %"SEL4_PRIu_word" of %"SEL4_PRIu_word"\n",
+           info->nodeID, info->numNodes);
+
+    printf("IO-MMU PT levels:       %"SEL4_PRIu_word"\n",
+           info->numIOPTLevels);
+
+    printf("initial thread domain:  %"SEL4_PRIu_word"\n",
+           info->initThreadDomain);
+
+    printf("root cnode size:        2^%"SEL4_PRIu_word" (%"SEL4_PRIu_word")\n",
+           info->initThreadCNodeSizeBits,
+           LIBSEL4_BIT(info->initThreadCNodeSizeBits));
+
+    printf("IPC buffer:             %p\n", info->ipcBuffer);
+
+    printf("shared pages:           ");
+    print_slot_reg_info(&info->sharedFrames);
+    printf("\n");
 
-    printf("Node %lu of %lu\n", (long)info->nodeID, (long)info->numNodes);
-    printf("IOPT levels:     %u\n", (int)info->numIOPTLevels);
-    printf("IPC buffer:      %p\n", info->ipcBuffer);
-    printf("Empty slots:     [%lu --> %lu)\n", (long)info->empty.start, (long)info->empty.end);
-    printf("sharedFrames:    [%lu --> %lu)\n", (long)info->sharedFrames.start, (long)info->sharedFrames.end);
-    printf("userImageFrames: [%lu --> %lu)\n", (long)info->userImageFrames.start, (long)info->userImageFrames.end);
-    printf("userImagePaging: [%lu --> %lu)\n", (long)info->userImagePaging.start, (long)info->userImagePaging.end);
-    printf("untypeds:        [%lu --> %lu)\n", (long)info->untyped.start, (long)info->untyped.end);
-    printf("Initial thread domain: %u\n", (int)info->initThreadDomain);
-    printf("Initial thread cnode size: %u\n", (int)info->initThreadCNodeSizeBits);
-    printf("List of untypeds\n");
-    printf("------------------\n");
-    printf("Paddr    | Size   | Device\n");
-
-    int sizes[CONFIG_WORD_SIZE] = {0};
-    for (int i = 0; i < CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS && i < (info->untyped.end - info->untyped.start); i++) {
```

Comment:
This ordering is important because it matches the order of caps in the root CNode.